etc/travis/Configuration.py

```diff
@@ -0,0 +1,50 @@
+import json
+import os
+import time
+
+from azure.mgmt.resource.resources.v2017_05_10.models import DeploymentMode
+
+
+class Configuration:
+    def __init__(self):
+        self.deployment_name = 'azuredeploy'
+
+        self.client_id = os.getenv('SPNAME')
+        self.secret = os.getenv('SPPASSWORD')
+        self.tenant_id = os.getenv('SPTENANT')
+        self.subscription_id = os.getenv('SPSUBSCRIPTION')
+        self.location = os.getenv('LOCATION', 'southcentralus')
+
+        self.ssh_key = os.getenv('SPSSHKEY')
+        if self.ssh_key is None:
+            with open('azure_moodle_id_rsa.pub', 'r') as sshkey_fd:
+                self.ssh_key = sshkey_fd.read()
+
+        self.resource_group = os.getenv('RESOURCEGROUP')
+        if self.resource_group is None:
+            self.resource_group = 'azmdl-travis-' + os.getenv('TRAVIS_BUILD_NUMBER', 'manual-{}'.format(time.time()))
+
+        self.deployment_properties = self.generate_properties()
+
+    def generate_properties(self):
+        with open('azuredeploy.json', 'r') as template_fd:
+            template = json.load(template_fd)
+
+        with open('azuredeploy.parameters.json', 'r') as parameters_fd:
+            parameters = json.load(parameters_fd)
+        parameters = parameters['parameters']
+        parameters['sshPublicKey']['value'] = self.ssh_key
```

Comment:
We do need to override the _artifactsLocation parameter here with the correct fork name and the branch name. Otherwise, only the top-level template file (azuredeploy.json) will be from this PR/commit, and all the nested templates will be pulled from Azure/Moodle:master, which is wrong.

Reply:
My apologies for not understanding the reason for that parameter before, that's the main reason why I asked why we need it. I'll fix it.

Reply:
That's OK. I should also probably have let you keep using bash, not rewriting the same things in Python anyway. Your fix looks good.

Reply:
There was a minor change to the logic, but the idea stays the same